providers/packetframe/packetframeProvider.go

```diff
@@ -0,0 +1,237 @@
+package packetframe
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+)
+
+// packetframeProvider is the handle for this provider.
+type packetframeProvider struct {
+	client      *http.Client
+	baseURL     *url.URL
+	token       string
+	domainIndex map[string]zone
+}
+
+var defaultNameServerNames = []string{
+	"ns1v4.packetframe.com",
+	"ns2v4.packetframe.com",
+}
+
+// newPacketframe creates the provider.
+func newPacketframe(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+	if m["apikey"] == "" {
```

Comment:
I haven't read the PacketFrame docs, but do they call this an API key or a token?  The JSON file keys should match what users see when reading the docs.

Reply:
Ah! Yeah that comes back from when I was doing this for packetframe v3 where it was called apikey